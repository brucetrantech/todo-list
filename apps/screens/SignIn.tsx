import { useState } from "react";
import { Image, StyleSheet, Text, TextInput, View } from "react-native";
import { YTButton, YTLayout } from "@/cores";
import contents from "@/commons/contents";
import images from "@/commons/images";
import screens from "@/commons/screens";
import themes from "@/commons/themes";
import apis from "@/storages/apis";

type SignInProps = {
	navigation: any;
};

export default function SignIn({ navigation }: SignInProps) {

	const [email, setEmail] = useState<string | undefined>(undefined);

	const onSignIn = () => {
		if (!email) return;
		apis.setCurrentUserByEmail(email)
			.then((result) => {
				if (!result) return;
				navigation.navigate(screens.HOME);
			})
	}

	return (
		<YTLayout>
			<View style={styles.layout}>
				<View style={styles.form}>
					<View style={styles.header}>
						<Text style={styles.title}>{contents.WELCOME_ONBOARD}</Text>
						<Image source={images.back} resizeMode="contain" />
					</View>
					<View>
						<TextInput
							placeholder={contents.ENTER_EMAIL}
							style={styles.input}
							value={email}
							onChangeText={setEmail}
						/>
						<View style={styles.line} />
						<View style={styles.line} />
						<View style={styles.line} />
						<YTButton
							title={contents.SIGN_IN}
							onPress={onSignIn}
							disabled={!email}
						/>
					</View>
				</View>
			</View>
		</YTLayout>
	)
}

const styles = StyleSheet.create({
	layout: {
		flex: 1,
		justifyContent: 'center',
	},
	form: {
		alignSelf: 'center',
		width: '100%',
	},
	header: { marginBottom: 20, alignItems: 'center' },
	title: {
		...themes.font.largeBold,
		color: themes.color.dark,
		marginBottom: 20,
	},
	line: {
		width: '100%',
		height: 12,
	},
	input: {
		width: '100%',
		paddingHorizontal: 20,
		paddingVertical: 12,
		borderRadius: 8,
		borderWidth: 1,
		borderColor: themes.color.secondaryNebula,
		borderStyle: 'solid',
		backgroundColor: themes.color.light,
	},
})