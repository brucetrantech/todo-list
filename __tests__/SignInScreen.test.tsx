/**
 * @format
 */

import 'react-native';
import React from 'react';
import SignIn from '@/screens/SignIn';

// Note: import explicitly to use the types shipped with jest.
import {it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('[Snapshot] SignIn screen renders correctly', () => {
  const navigation = {
    navigate: jest.fn(),
  }
  const tree = renderer.create(<SignIn navigation={navigation} />).toJSON();
  expect(tree).toMatchSnapshot();
});
