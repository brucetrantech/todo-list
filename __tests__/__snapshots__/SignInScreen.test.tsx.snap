// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Snapshot] SignIn screen renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "flex": 1,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../apps/assets/images/background.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RCTSafeAreaView
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "paddingHorizontal": 24,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          [
            {
              "flex": 1,
            },
            {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              {
                "alignSelf": "center",
                "width": "100%",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "marginBottom": 20,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#000000",
                    "fontFamily": "Poppins-Medium",
                    "fontSize": 20,
                    "fontWeight": "800",
                    "marginBottom": 20,
                  }
                }
              >
                Welcome Onboard!
              </Text>
              <Image
                resizeMode="contain"
                source={
                  {
                    "testUri": "../../../apps/assets/images/back.png",
                  }
                }
              />
            </View>
            <View>
              <TextInput
                autoCapitalize="none"
                onChangeText={[Function]}
                onSubmitEditing={[Function]}
                placeholder="Enter your email"
                placeholderTextColor="#D7D7D7"
                style={
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#D7D7D7",
                    "borderRadius": 8,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "color": "#000000",
                    "paddingHorizontal": 20,
                    "paddingVertical": 12,
                    "width": "100%",
                  }
                }
              />
              <View
                style={
                  {
                    "height": 30,
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              />
              <View
                style={
                  {
                    "height": 12,
                    "width": "100%",
                  }
                }
              />
              <View
                style={
                  {
                    "height": 12,
                    "width": "100%",
                  }
                }
              />
              <View
                style={
                  {
                    "height": 12,
                    "width": "100%",
                  }
                }
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#D7D7D7",
                    "borderColor": "#D7D7D7",
                    "borderRadius": 8,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "minHeight": 42,
                    "minWidth": 42,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "Poppins-Medium",
                        "fontSize": 14,
                        "fontWeight": "500",
                      },
                      {
                        "color": "#3E3E3E",
                      },
                    ]
                  }
                >
                  Sign In
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
